tests/python/openassetio_mediacreation/test_specifications.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The Foundry Visionmongers Ltd
+"""
+Tests Specifications include required traits.
+"""
+
+# pylint: disable=no-self-use
+# pylint: disable=invalid-name
+# pylint: disable=missing-class-docstring
+
+import inspect
+
+import pytest
+
+from openassetio_mediacreation.traits.usage import EntityTrait, RelationshipTrait
+from openassetio_mediacreation import specifications
+
+
+class Test_specification_traits:
+    def test_contain_usage_trait(self):
+        """
+        All specifications should contain one trait from the usage namespace
+        so that their purpose can be determined at runtime.
+        """
+        for cls in self.__specifications():
+            # Ensure exactly one is present
+            assert (EntityTrait.kId in cls.kTraitSet) != (RelationshipTrait.kId in cls.kTraitSet)
+
+    def test_traits_constructible(self):
```

Comment:
[plan] Good idea, but unfortunate that we can't trust that TraitGen did this validation already. Worth a TraitGen issue to look into?

Reply:
Depends in you want to have to provide builds of all dependent packages (eg: when composing traits/specs from another package)?